pkgs/by-name/to/tomlplusplus/package.nix

```diff
@@ -0,0 +1,67 @@
+{
+  cmake,
+  fetchFromGitHub,
+  fetchpatch2,
+  glibcLocales,
+  lib,
+  meson,
+  ninja,
+  nix-update-script,
+  pkg-config,
+  stdenv,
+  testers,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "tomlplusplus";
+  version = "3.4.0";
+
+  src = fetchFromGitHub {
+    owner = "marzer";
+    repo = "tomlplusplus";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-h5tbO0Rv2tZezY58yUbyRVpsfRjY3i+5TPkkxr6La8M=";
+  };
+
+  patches = [
+    # https://github.com/marzer/tomlplusplus/pull/233
+    (fetchpatch2 {
+      name = "tomlplusplus-install-example-programs.patch";
+      url = "https://github.com/marzer/tomlplusplus/commit/8128eb632325d1820f4d17dd8250dcda6ab07743.patch";
+      hash = "sha256-7m2P+e1/OASHrzm9LSy6RnayS/kGxFC82xOyGBGXeG0=";
+    })
+  ];
+
+  nativeBuildInputs = [
+    meson
+    cmake
+    ninja
+    pkg-config
+  ];
+
+  checkInputs = [
+    glibcLocales
+  ];
+
+  doCheck = true;
+  mesonFlags = [
+    "-Dbuild_tests=${lib.boolToString finalAttrs.doCheck}"
+    "-Dbuild_examples=true"
+  ];
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests.pkg-config = testers.hasPkgConfigModules {
+      package = finalAttrs.finalPackage;
+    };
+  };
+
+  meta = with lib; {
```

Comment:
no `meta.mainProgram` ?

Reply:
Not really. This is primarily a C++ library and the binaries that we package here are just examples from upstream. They aren't actually supported tools.

Once I fix upstream's cmake generation I can at least bring back split outputs for this package so users dont have to pull in these examples.